gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -285,12 +285,21 @@ MacroExpander::expand_invoc (AST::MacroInvocation &invoc, bool has_semicolon)
   last_def = rules_def;
 
   if (rules_def->is_builtin ())
-    fragment
-      = rules_def->get_builtin_transcriber () (invoc.get_locus (), invoc_data);
-  else
-    fragment = expand_decl_macro (invoc.get_locus (), invoc_data, *rules_def,
-				  has_semicolon);
+    {
+      auto maybe_fragment
+	= rules_def->get_builtin_transcriber () (invoc.get_locus (),
+						 invoc_data);
 
+      if (maybe_fragment.has_value ())
+	fragment = maybe_fragment.value ();
+      else
+	fragment = AST::Fragment::create_empty ();
```

Comment:
not functional enough >:)

```suggestion
      fragment
	= rules_def->get_builtin_transcriber () (invoc.get_locus (),
						 invoc_data)
	.value_or(AST::Fragment::create_empty());
```

Reply:
(if you go with this I guess we should also remove the extra curly brackets in the `if` and `else` blocks)

Reply:
done